neqo-crypto/src/secrets.rs

```diff
@@ -44,23 +45,16 @@ impl From<SSLSecretDirection::Type> for SecretDirection {
 #[derive(Debug, Default)]
 pub struct DirectionalSecrets {
     // We only need to maintain 3 secrets for the epochs used during the handshake.
-    secrets: [Option<SymKey>; 3],
+    secrets: EnumMap<Epoch, Option<SymKey>>,
```

Comment:
Does this need to be `Option<SymKey>` or just `EnumMap<Epoch, SymKey>` ?  Because we previously needed each to be optionally present, but the map already supports that.